### PR TITLE
feat(components): Add `Form` component

### PR DESCRIPTION
**Pull request by edizcelik**
### Related Issues

- #22

### CODE REVIEW

File: src/components/form/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  });
+
+  return (
+    <form
+      className={containerClassName}
+      data-testid={testid}
+      onChange={handleFormChange}
+      onSubmit={handleSubmit}>
+      {Boolean(formError) && (
+        <p data-testid={`${testid}.form-error-message`}>{formError}</p>
```
Comment (mucahit) [inline src/components/form/Form.tsx]: Let's add a class name to the error message so we can customize the style easily 
```suggestion
        <p className={"form-error-message"} data-testid={`${testid}.form-error-message`}>{formError}</p>
```

**Review by mucahit — APPROVED** [review]

File: src/components/form/Form.tsx (hunk 1 above)
Comment (edizcelik) [reply src/components/form/Form.tsx]: Done! I also added basic styles.

**Review by bilgeyucel — APPROVED** [review]

File: src/components/form/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  } = props;
+  const [isPristine, setPristine] = useState(true);
+  const containerClassName = classNames("form", customClassName);
+  const formError = generateErrorMessage(errorInfo, {
+    knownErrorKeys,
+    skipTypes: skipErrorTypes
+  });
+
+  return (
+    <form
```
Comment (burakguneli) [inline src/components/form/Form.tsx:42]: Maybe we can add `autocomplete` attribute with default value set to `"off"`. I needed it a couple of times but I'm not sure if we need to add this to project starter

**Review by burakguneli — APPROVED** [review]

File: src/components/form/Form.tsx (hunk 2 above)
Comment (edizcelik) [reply src/components/form/Form.tsx:42]: I think we can add it as a prop and it can default to `"off"`. What do you think ?

Comment (burakguneli) [reply src/components/form/Form.tsx:42]: I agree

**Pull request merged**